Removing jQuery

sebworks opened the pull request:
Removing jQuery and it's associated files.

## Additions

- Added Carousel code ( this code is just minor modifications to the SwipeJs code ).

## Removals

-  Removing jQuery and it's associated files.


## Changes

- Modified less file to be current with DOM modifications.

## Testing
- Run 'gulp build`
-  Visit `http://localhost:8000/about-us/the-bureau/bureau-structure/` and it should perform as expected.

## Review

- @cfpb/cfgov-frontends 

## Notes

- `SwipeJS` currently needs some modifications, so I've included that code internally. It might make sense just to fork it.
- We still load jQuery from the CDN because it's used by the analytics team and in other apps.

## Todos

-

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

jimmynotjim commented on cfgov/unprocessed/js/modules/Carousel.js:
```diff
@@ -0,0 +1,759 @@
+*/
+
+// if the module has no dependencies, the above pattern can be simplified to
+;(function (root, factory) {
```
This is all copied code right? None of it's custom to our project?

jimmynotjim commented on cfgov/unprocessed/js/routes/about-us/the-bureau/bureau-structure/index.js:
```diff
@@ -48,44 +45,33 @@ function _initExpandables() {
-  _slider = new ContentSlider( sliderElem ).init( 1 );
-  _slider.addEventListener( 'afterChange', _sliderAfterChangeHandler )
+function _createCarousel() {
+  	_updateHeight( 0, _orgChartBranches.firstElementChild );
```
Minor: L:53 & 54 have an extra level of indentation.

jimmynotjim commented on cfgov/unprocessed/js/routes/about-us/the-bureau/bureau-structure/index.js:
```diff
@@ -48,44 +45,33 @@ function _initExpandables() {
-  // Show org chart branches.
-  $( '.org-chart_branches' ).show();
+  if ( _swipe ) {
+  		_orgChartBranches.style = ''
```
Fix indentation and close with a `;`

jimmynotjim commented on cfgov/unprocessed/css/pages/bureau-structure.less:
```diff
@@ -12,88 +12,27 @@
     background: @gray-10;
     border-top:1px solid @node-border-color;
 
+    .u-hide-unless-mobile {
```
`.u-show-on-mobile` already [exists in cf-core](https://github.com/cfpb/capital-framework/blob/canary/src/cf-core/src/cf-utilities.less#L262). Use that instead.

jimmynotjim commented on cfgov/unprocessed/css/pages/bureau-structure.less:
```diff
@@ -116,6 +55,24 @@
+    .respond-to-max( @bp-xs-max, {
+        .swipe {
+          overflow: hidden;
+          visibility: hidden;
```
Is the visibility over-written by the JS? If so, do we have a fallback for non-js situations?

sebworks commented:
@jimmynotjim, I modified the PR to do the following: 

- Removed `SwipeJS` because swiping isn't necessary.
- Added a Bureau Structure module to handle transitioning in / out org chart branches.  I opted not to use the utility classes to handle the transition and instead relied upon JS. 
- Modified some of the names of Bureau Structure templates and classes.

jimmynotjim commented on cfgov/unprocessed/css/pages/bureau-structure.less:
```diff
@@ -116,9 +46,37 @@
+        position: relative;
+
+        .respond-to-min( @bp-sm-min, {
+            margin: @margin / 2 0;
```
Minor, I'd wrap `@margin / 2` in parens just so it's clear that's the top/bottom and the 0 is left right. It should also probably be converted to `em`s

jimmynotjim commented on cfgov/unprocessed/css/pages/bureau-structure.less:
```diff
@@ -116,9 +46,37 @@
+        } );
+
+        .respond-to-max( @bp-xs-max, {
+            margin-bottom: @margin;
```
This should also be in `em`s right?

sebworks replied:
The variable is in `ems`.

jimmynotjim replied:
Whoops, sorry, I thought @margin was a px value.

jimmynotjim commented on cfgov/unprocessed/css/pages/bureau-structure.less:
```diff
@@ -138,7 +96,7 @@
         }
 
-        &:first-child {
+        &:nth-child(2n) {
```
Minor, add spaces inside parens

jimmynotjim commented on cfgov/unprocessed/css/pages/bureau-structure.less:
```diff
@@ -207,27 +142,85 @@
 
-    .link_text {
-       border: none;
+    .org-chart_nav-item {
```
How is this different from the `li` above? Related, how is the same class on L:177 different from this one? Lastly, does it need to be nested here? By BEM naming it should already be a child of `org-chart` and shouldn't need any more specificity.

sebworks replied:
So the `.org-chart_nav-item` class is targeting navigation buttons which appear within two separate DOM structures.  It's nested because we want to display the buttons differently when they appear within `org chart nav`.

sebworks replied:
if you make it a modifier then you have to reset some of the styles.

jimmynotjim commented on cfgov/unprocessed/css/pages/bureau-structure.less:
```diff
@@ -351,7 +343,7 @@
         padding-top: 0;
-        margin-top: @margin;
         border: none;
+        margin-top: @margin;
```
Also convert this value to `em`s

jimmynotjim commented on package.json:
```diff
@@ -72,8 +72,7 @@
     "es5-sham": "./node_modules/es5-shim/es5-sham.min.js",
-    "jquery": "./node_modules/jquery/dist/jquery.js",
-    "slick": "./node_modules/slick-carousel/slick/slick.js"
+    "jquery": "./node_modules/jquery/dist/jquery.js"
```
Should jQuery be removed here?

sebworks replied:
So we still need it for the CDN fallback in `base.html`.

virginiacc commented on cfgov/unprocessed/js/modules/BureauStructure.js:
```diff
@@ -0,0 +1,249 @@
+      msie: 'ms'
+    };
+
+    return vendors[match[0]];
```
Should this return an empty string if no match/vendor is found?

virginiacc commented:
This looks great! The only issue I'm seeing is that the mobile transition doesn't seem to be working in IE11, although it does work in IE10.

sebworks commented:
> This looks great! The only issue I'm seeing is that the mobile transition doesn't seem to be working in IE11, although it does work in IE10.

Fixed in https://github.com/cfpb/cfgov-refresh/pull/2550/commits/c211e966206ada4c9ae6abe9e62f16bcedd9431e.

sebworks commented:
@jimmynotjim, @virginiacc  please take a look at this again. I've made changes to bring it inline with the migration of the Bureau Structure to Wagtail.

jimmynotjim commented:
The code looks good. I can't test it in IE11 at this moment, but hopefully @virginiacc can.

virginiacc commented:
👍 Tested in IE11. Everything looks great!

anselmbradford commented:
Found jquery reference [here](https://github.com/cfpb/cfgov-refresh/blob/c90d799466bffcf8ef3058c81f10787fb95f59ba/cfgov/unprocessed/js/modules/ExternalSite.js#L52). Is that file still used?

sebworks commented:
hmmm, that should be removed. I'll create a PR for that.
